src/Redux/Books/books.js

```diff
@@ -16,32 +21,72 @@ const bookReducer = (state = initialState, action) => {
       return {
         ...state,
         isLoading: false,
-        books: [...state.books, payload],
+        msg: payload,
       };
     case REMOVE_BOOK:
       return {
         ...state,
         isLoading: false,
-        books: state.books.filter((book) => book.id !== payload),
+        books: state.books.filter((book) => book.id !== payload.id),
+        msg: payload.msg,
+      };
+    case SHOW_BOOKS:
+      return {
+        ...state,
+        isLoading: false,
+        books: payload,
       };
     default:
       return state;
   }
 };
 
 // Actions
-export const addBook = (values) => (dispatch) => {
-  dispatch({
-    type: ADD_BOOK,
-    payload: values,
-  });
+export const showBook = () => async (dispatch) => {
+  try {
+    const { data } = await axios.get(baseUrl);
```

Comment:
- [ ] Everything in your project is working as expected. However, based on Project requirements you have to use createAsyncThunk for creating thunk action creators:  
```
// This code is just an example

import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
);
```

Please make sure that thunk action creator for fetching books is  created

* It should return a promise.
* It should fetch books list from API.
* It should automatically dispatch an action type that will pass books list to reducer and updates state.

Reply:
Hello @ShoiraTa,
I didn't use createSlice for my project without createSlice how I can use createAsyncThunk in my project?